### DISCUSSION TITLE
“MeshSideSetGenerator" without “SideSetsBetweenSubdomainsGenerator”

### DISCUSSION THREAD

**Discussion by styyokuda** [idaholab/moose#20829] [Q&A Meshing] [answered]
All,
I would like to set boundaries using “MeshSideSetGenerator” as seen from the example of test/tests/meshgenerators/mesh_side_set_generator/mesh_side_set_generator.i (please see below).

I am trying to use the boundary name generated by Cubit for "boundaries =" in “MeshSideSetGenerator” so that I don’t have to use “SideSetsBetweenSubdomainsGenerator” but it is “not” working.  

Do I have to use “SideSetsBetweenSubdomainsGenerator” to generate the boundary (like “center_side_set” in mesh_side_set_generator.i below) which is inputted in “MeshSideSetGenerator”?

Please help.

S. Thomas

(test/tests/meshgenerators/mesh_side_set_generator/mesh_side_set_generator.i)
[Mesh]
.
.
.
  [./center_side_set]
    type = SideSetsBetweenSubdomainsGenerator
    input = right_block
    primary_block = left_block
    paired_block = right_block
    new_boundary = center_side_set
  []

  [./center_mesh]
    type = MeshSideSetGenerator
    input = center_side_set
    boundaries = center_side_set
    block_id = 10
    block_name = center_mesh
  []
[]

**Comment by GiudGiud** [comment, marked as answer]
Hello

If Cubit already added the sideset you shouldnt need to recreate it using MeshGenerators. 
If you open your mesh file in Paraview, and click the advanced information parameters, do you see the sideset in your mesh?

Guillaume

**Comment by styyokuda** [comment]
Thank you very much Guillaume,
In “Information” Tab in test/tests/meshgenerators/mesh_side_set_generator/mesh_side_set_generator_out.e,
I see:
“center_mesh” in Element Blocks,
“center_side_set” in Side Sets,
“Unnamed set ID: 6 in Node Sets.

So, “center_side_set” in Side Sets was generated by SideSetsBetweenSubdomainsGenerator, and “center_mesh” in Element Blocks was generated by MeshSideSetGenerator.

The mesh generated by Cubit I have has only the mesh information in the Node Sets, not in Element Blocks nor Side Sets. So, that is the problem, isn’t it?

Can Cubit generate a mesh with like “center_mesh” in Element Blocks and “center_side_set” in Side Sets?

Please help. 
S. Thomas

**Reply by GiudGiud** [reply]
So MOOSE will convert automatically node sets to side sets if you pass the right parameter to the Mesh block. See this page
https://mooseframework.inl.gov/source/mesh/MooseMesh.html

Then you can use the Cubit mesh and still have access to the sideset even though the cubit mesh only came with a node set

**Reply by styyokuda** [reply]
Thank you very much Guillaume,
I don't know how to use
construct_side_list_from_node_list=false

What the MOOSE class is it in?
Do you have an example?
Please help.
S. Thomas

**Reply by GiudGiud** [reply]
put it in the [Mesh] block
like this
```
[Mesh]
  construct_side_list_from_node_list = true
  ...
[]

**Reply by styyokuda** [reply]
Thank you very much Guillaume,
Now I see Side sets.
I believe that "MeshSideSetGenerator" still needs to be used because "MeshSideSetGenerator" generates "blocks" for a model as Side Sets are for boundaries.
Please help.
S. Thomas

**Reply by GiudGiud** [reply]
no you dont need to use that.
This is for a very narrow use where you want to solve on lower dimensional elements on sidesets. I ll try to rename this object if I m right

**Reply by styyokuda** [reply]
Thank you very much Guillaume,
I got confused. To me, Side Sets are boundaries used to set boundary conditions and so on. "MeshSideSetGenerator" generates elements which are one dimension lower than other elements (one dimensional elements for 2D model). So, I can use these lower dimensional elements  for diffusion and heat equations with sources by assigning material properties.
Please hep.
S. Thomas

**Reply by GiudGiud** [reply]
no do not use this generator. This is for solving equations on the lower dimensional mesh. It's very poorly named. 

Sidesets are used in the regular workflow for boundary conditions.
